test/cocotb/osdtestlib/debug_interconnect.py

```diff
@@ -245,75 +321,73 @@ def write_register(self, dut, destination, source, reg_width, address,
             address:        address of the register the new value will be
                             written to.
             content:        value which will be written into the specified
                             register.
-            response:       buffer variable which the response of the dut will
-                            be stored in.
 
             Returns:
             -
 
         """
 
         # depending on how many data words are to be written into the memory
-        if reg_width == 16:
-
-            subtype = self.PACKET_SUBTYPE_REQ_WRITE_REG_16
+        if word_width == 16:
+            subtype = PACKET_SUBTYPE_REQ_WRITE_REG_16
             words = 1
-
-        elif reg_width == 32:
-
-            subtype = self.PACKET_SUBTYPE_REQ_WRITE_REG_32
+        elif word_width == 32:
+            subtype = PACKET_SUBTYPE_REQ_WRITE_REG_32
             words = 2
-
-        elif reg_width == 64:
-
-            subtype = self.PACKET_SUBTYPE_REQ_WRITE_REG_64
+        elif word_width == 64:
+            subtype = PACKET_SUBTYPE_REQ_WRITE_REG_64
             words = 3
-
-        elif reg_width == 128:
-
-            subtype = self.PACKET_SUBTYPE_REQ_WRITE_REG_128
+        elif word_width == 128:
+            subtype = PACKET_SUBTYPE_REQ_WRITE_REG_128
             words = 4
-
         else:
 
             # tell the user that an unsupported register width was chosen
-            raise TestFailure("Invalid register width parameter")
+            raise RegAccessFailedException(message="An invalid register width\
+            parameter was chosen!")
 
-
-        payload = []
-        payload.append(address)
+        payload = [ address ]
 
         for i in range(0, words):
             # how many times must the 16 bit mask be shifted to the left?
-            shift_count = words-(i+1)
-            payload.append(((content & (0xFFFF << (16*shift_count)))) >> \
-                           (16*shift_count))
-
-        packet = DiPacket(dest = destination, source = source,
-                          packet_type = self.PACKET_TYPE_REG,
-                          packet_subtype = subtype, payload = payload)
+            shift_count = words - (i + 1)
+            payload.append(((content & (0xFFFF << (16 * shift_count)))) >> \
+                           (16 * shift_count))
 
+        tx_packet = DiPacket(destination=destination, source=source,
+                             packet_type=PACKET_TYPE_REG,
+                             packet_subtype=subtype, payload=payload)
 
-        yield self.send_packet(dut = dut, DiPacket = packet)
+        yield self.send_packet(dut=dut, packet=tx_packet)
 
         # read the response of the debug module
         response = []
-        yield self.receive_packet(dut = dut, response = response)
+        yield self.receive_packet(dut=dut, response=response)
+
+        # display the response
+        dut._log.info("Answer of the dut:")
+        dut._log.info('[{}]'.format(', '.join(hex(x) for x in response)))
```

Comment:
If you add a DiPacket.__str__() method to the DiPacket class, you can save yourself from repeatedly writing code which nicely prints out a packet. Instead, you can simply write `dut._log.info(str(your_packet))`. __str__ is a special function like toString() in Java, find the documentation here: https://docs.python.org/3/reference/datamodel.html#object.__str__